sqlmesh/core/model/meta.py

```diff
@@ -64,6 +67,8 @@ class ModelMeta(_Node):
     references: t.List[exp.Expression] = []
     physical_schema_override: t.Optional[str] = None
     table_properties_: t.Optional[exp.Tuple] = Field(default=None, alias="table_properties")
+    physical_properties_: t.Optional[exp.Tuple] = Field(default=None, alias="physical_properties")
```

Comment:
https://github.com/TobikoData/sqlmesh/tree/main/sqlmesh/migrations

Reply:
Isn't these fields meant for Pydantic field parsing?
My goal is to be able to parse:
- table_properties ("deprecated")
- physical_properties (the new table_properties)
- virtual_properties (the new properties for virtual layer)

Should that append with a custom `Field` to factor `table_properties`and `physical_properties`.

> can we remove this and do a migration to move all physical_properties in the database to virtual_properties?
Do you mean all `table_properties` to `physical_properties` to migrate?

Reply:
so i imagine pydantic will only have two fields, physical/virtual_properties. we migrate in the state all usage of table_properties -> physical properties.

at parse time, if we detect table_properties, we log a message and remap that to physical_properties

Reply:
I don't think we need to migrate anything. Can't we just add a `table_properties` alias to `physical_properties_`?

Reply:
If not, we can also handle this in the root model validator. And just pop `table_properties`, emit a warning, and then set it as `physical_properties`

Reply:
In any case, only `physical_properties` should stay, and no updates should be necessary to the `EngineAdapter` layer.

Reply:
I also don't think a migration is needed or maybe I'm missing?
@izeigerman, in [this commit](https://github.com/TobikoData/sqlmesh/pull/2561/commits/32bdcbd10bcf5198c5a8925f651cb40e7c51b52f) I've updated the Pydantic parsing to use both with the related warning. Isn't it what you suggest?
Regarding `EngineAdapter`, I think we could indeed skip the `physical` / `virtual` naming and skip to the actual materialization: `table_properties` / `view_properties` WDYT?

Reply:
In [this commit](https://github.com/TobikoData/sqlmesh/pull/2561/commits/1632b9866fd1935e0b57381c359d318c3611423b) I tried to set proper boundaries with physical/virtual until the evaluator that translate those into table/view (depending on the actual chosen materialization).

Reply:
Hey @Kayrnt, after additional consideration, I've realized that you do need the migration after all. Since we're getting rid of the `table_properties` field altogether we do need to reflect this in the existing state.

The fact that you had to change the `fixtures/migrations/snapshots.json` file confirms this suspicion. If we don't migrate, the next time users run the `plan` command after migration, it will show the diff because the field name has changed.

You can test this by first initializing a project with duckdb using a previous SQLMesh version and then upgrading to your current branch and running the plan command.

Sorry about the confusion initially, I really hoped we could avoid it.